### PR TITLE
Create 033_tip_dynamic_env_variables.md

### PR DESCRIPTION
**Pull request by yelhouti**
as discussed here: https://github.com/jhipster/generator-jhipster/pull/10538

### CODE REVIEW

File: tips/033_tip_dynamic_env_variables.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+layout: default
+title: Dynamic environement variables in the front end
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:3]: spelling, should be "environment"

File: tips/033_tip_dynamic_env_variables.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+sitemap:
+priority: 0.1
+lastmod: 2020-07-01T10:50:00-00:00
+---
+# Dynamic environement variables in the front end
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:8]: spelling, should be "environment"

File: tips/033_tip_dynamic_env_variables.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+__Tip submitted by [@yelhouti](https://github.com/yelhouti)__
+
+Let's say you need to update the value of a variable in the front end after your code have been compiled. (ex: your IdP Url, the email to use for contacts...)
+
+One way is to have it as part of your application.yml and have the backend return it to the front using a new custom endpoint, the same way we do in: `AuthInfoResource.java` when using OAuth2.
+
+Another way that removes the need for this endpoint and offers better flexibility and less code is to have a new file called `env.js` that looks like this:
+
+```
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:18]: Is it possible in Jekyll to enable syntax highlighting for code blocks?

File: tips/033_tip_dynamic_env_variables.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    <!-- jhipster-needle-add-resources-to-root - JHipster will add new resources here -->
+    <script src="env.js"></script>
+```
+and we tell webpack to copy it, as is, with the packaged code:
+```
+// jhipster-needle-add-assets-to-webpack - JHipster will add/remove third-party resources in this array
+{ from: './<%= MAIN_SRC_DIR %>env.js', to: 'env.js' },
+```
+
+We recommand adding the file to the `.eslintignore.ejs` for the clean syntax:
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:44]: spelling, should be "recommend"

File: tips/033_tip_dynamic_env_variables.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+The code below creates a global `__env` variable if not already declared.
+
+it can be accessed in your angular files but we recommend exposing it through constants.ts like this:
+```
+// @ts-ignore
+// export const MY_DYNAMIC_VARIABLE = window.__env.myDynamicVariable;
+```
+When using kubernetes, a file with this format can be mounted as a configMap, this is why we keep such a simple key value syntax.
+
+Now we need to make sur that `index.html` loads it by adding the script tag in the `<head>` like this:
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:32]: spelling, should be "sure"

File: tips/033_tip_dynamic_env_variables.md (hunk 3 above)
Comment (yelhouti) [reply tips/033_tip_dynamic_env_variables.md:18]: Aucune idée, jamais utilisé Jekyll, dit moi quoi faire et je le fais sans problème.

Comment (ruddell) [reply tips/033_tip_dynamic_env_variables.md:18]: You can specify the language next to the ``` like [here](https://github.com/jhipster/jhipster.github.io/commit/92195598afecdd1b0675a98abbfa8c96351646f6#diff-8f89e32c752d5048d1aafce1d01f05b0R71) for example.  [Supported languages](https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers) include typescript and javascript

**Comment by yelhouti** [comment]
@gmarziou done :) thank you @ruddell

File: tips/033_tip_dynamic_env_variables.md [hunk 6]
```diff
@@ -15,28 +15,28 @@ One way is to have it as part of your application.yml and have the backend retur
...
 window.__env.myDynamicVariable = 'http://127.0.0.1:8090';
 ```
 
 The code below creates a global `__env` variable if not already declared.
 
 it can be accessed in your angular files but we recommend exposing it through constants.ts like this:
-```
-// @ts-ignore
-// export const MY_DYNAMIC_VARIABLE = window.__env.myDynamicVariable;
+```typescipt
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:26]: you forgot an "r"

File: tips/033_tip_dynamic_env_variables.md [hunk 7]
```diff
@@ -15,28 +15,28 @@ One way is to have it as part of your application.yml and have the backend retur
...
 Now we need to make sure that `index.html` loads it by adding the script tag in the `<head>` like this:
-```
+```html
     ...
     <!-- jhipster-needle-add-resources-to-root - JHipster will add new resources here -->
     <script src="env.js"></script>
 ```
 and we tell webpack to copy it, as is, with the packaged code:
-```
+```javascipt
```
Comment (gmarziou) [inline tips/033_tip_dynamic_env_variables.md:39]: you forgot an "r"

**Comment by yelhouti** [comment]
I am really bad sorry

**Pull request merged**